src/Nethermind/Nethermind.Core/Transaction.cs

```diff
@@ -72,40 +75,49 @@ void GenerateHash()
             }
             set
             {
-                if (_preHash.Length > 0)
+                lock (this)
```

Comment:
In theory you shouldn't lock on anything public, but as we don't want to bloat the object we probably don't have anything better